src/mqueryfront/src/QueryPage.js

```diff
@@ -55,6 +56,30 @@ class QueryPage extends Component {
         });
     }
 
+    componentDidUpdate(prevProps, prevState) {
+        if (
+            this.props.match.path === "/" &&
+            this.props.location.state &&
+            this.props.location.state.fromNavigation &&
+            (typeof prevProps.location.state === "undefined" ||
+                typeof prevProps.location.state.fromNavigation === "undefined")
+        ) {
+            //reset page to unitial state
+            this.setState({
+                mode: "query",
```

Comment:
this is the same state as in the constructor. maybe move this to some variable like `DEFAULT_STATE`?

Reply:
Good idea!. I'll use INITIAL_STATE constant.

Reply:
By the way, it wasn't exactly the same piece of code. In constructor the `qhash` variable was used for `qhash` attribute.

Reply:
INITIAL_STATE usage in contructor will be:
```
        let initialState = { ...INITIAL_STATE };
        if (this.props.match.params.hash) {
            initialState.qhash = this.props.match.params.hash;
        }
        this.state = initialState;

```